packages/good-design/src/apps/bridge/MicroBridge.tsx

```diff
@@ -0,0 +1,232 @@
+
+  const [sourceChain, setSourceChain] = useState<"fuse" | "celo">("fuse");
+
+  const targetChain = sourceChain === "fuse" ? "celo" : "fuse";
+  const balanceWei = useBalanceHook(sourceChain);
+
+  let { isValid, reason } = useCanBridge(sourceChain, inputWei);
+  const hasBalance = Number(inputWei) <= Number(balanceWei);
+  const isTargetValid = !enableTarget || isAddressValid(target || "");
+  const isValidInput = isValid && hasBalance && isTargetValid;
+  reason = reason || (!hasBalance && "balance") || (!isTargetValid && "target") || "";
+
+  const toggleChains = useCallback(() => {
+    setSourceChain(targetChain);
+    onSetChain && onSetChain(targetChain);
+  }, [setSourceChain, onSetChain, targetChain]);
+
+  const triggerBridge = useCallback(async () => {
+    setBridging(true);
+    try {
+      await onBridge(inputWei, sourceChain, enableTarget ? target : undefined);
+    } catch (e) {
+      setBridging(false);
+    }
+  }, [setBridging, onBridge, inputWei, sourceChain, enableTarget, target]);
+
+  useEffect(() => {
+    if (
+      ["Mining", "PendingSignature", "Success"].includes(bridgeStatus?.status || "") &&
+      (["None", "Fail", "Exception", "Success"].includes(selfRelayStatus?.status || "") &&
+        ["None", "Fail", "Exception", "Success"].includes(relayStatus?.status || "")) === false
+    ) {
+      setBridging(true);
+    } else {
+      setBridging(false);
+    }
+  }, [relayStatus, bridgeStatus, selfRelayStatus]);
+
+  return (
+    <Box>
+      <Flex direction="row" justifyContent={"space-between"}>
+        <Box>
+          <Text textTransform="capitalize">{sourceChain}</Text>
+        </Box>
+        <Switch boxSize="8" size="lg" isChecked={sourceChain === "fuse"} onToggle={toggleChains} />
+        <Box>
+          <Text textTransform="capitalize">{sourceChain === "fuse" ? "Celo" : "Fuse"}</Text>
+        </Box>
+      </Flex>
+      <TokenInput
+        balanceWei={balanceWei}
+        decimals={2}
+        onChange={setInput}
+        minAmountWei={limits?.[sourceChain]?.minAmount?.toString()}
+      />
+      <FormControl isInvalid={!!reason}>
+        <FormControl.ErrorMessage leftIcon={<WarningOutlineIcon variant="outline" />}>
+          {reason}
+        </FormControl.ErrorMessage>
+      </FormControl>
+      <Box mt="5">
+        <Checkbox isChecked={enableTarget} value="" onChange={() => setEnableTarget(!enableTarget)}>
+          Send to a different address
+        </Checkbox>
+        <AddressInput mt="2" onChange={setTarget} display={enableTarget ? "inherit" : "none"} />
+      </Box>
+      <Box mt="5">
+        <HStack>
+          <Text w={{ base: "1/2", sm: "1/4" }}>Fee</Text>
+          <Text w={{ base: "1/2", sm: "1/4" }}>{expectedFee} G$</Text>
+        </HStack>
+        <HStack>
+          <Text w={{ base: "1/2", sm: "1/4" }}>Expected to receive </Text>
+          <Text w={{ base: "1/2", sm: "1/4" }}>{(Number(inputWei) / 100 - Number(expectedFee)).toFixed(2)} G$</Text>
```

Comment:
all calculations could be moved to a single useMemo returning some data interface